### PR TITLE
CI: Local and Deployment Project Tests

### PR DESCRIPTION
**Pull request by fahreddinozcan**
This PR introduces a test pipeline, which also builds & deploys projects to various platforms and test on production. The Platform tests introduced with this PR:
- Cloudflare
  - [x] Deployment
  - [x] Local
- Next.js
  - [x] Deployment
  - [x] Local

### CODE REVIEW

**Comment by fahreddinozcan** [comment]
The pipeline actually works now as you can see in this [action](https://github.com/upstash/ratelimit-js/actions/runs/11204073489). However, this might still break sometimes, but that's due to ratelimit tests being flaky.

**Comment by ytkimirti** [comment]
Is fixing the flaky tests something trivial @fahreddinozcan?

**Comment by fahreddinozcan** [comment]
> Is fixing the flaky tests something trivial @fahreddinozcan?

not really sure. maybe we should open a new ticket for that. wdyt @CahidArda

**Comment by CahidArda** [comment]
> > Is fixing the flaky tests something trivial @fahreddinozcan?
> 
> not really sure. maybe we should open a new ticket for that. wdyt @CahidArda

yes, it's not that trivial. we can open another task.

File: examples/nextjs/app/api/route.ts [hunk 1]
```diff
@@ -6,9 +6,15 @@ import { waitUntil } from '@vercel/functions';
...
+const redis = new Redis({
+  url: process.env.UPSTASH_REDIS_REST_URL!,
+  token: process.env.UPSTASH_REDIS_REST_TOKEN!,
+})
+
 // Create a new ratelimiter
 const ratelimit = new Ratelimit({
-  redis: Redis.fromEnv(),
+  // @ts-ignore
+  redis,
```
[the remark this answers is not in the document]
Comment (CahidArda) [reply examples/nextjs/app/api/route.ts:17]: it's an issue in local builds, there is a type mismatch when you build locally and use it

File: examples/cloudflare-workers/package.json [hunk 2]
```diff
@@ -14,5 +16,7 @@
   "dependencies": {
     "@upstash/ratelimit": "^1.0.3",
     "@upstash/redis": "^1.20.6"
-  }
-}
+  },
+  "module": "index.ts",
+  "type": "module"
```
Comment (CahidArda) [inline examples/cloudflare-workers/package.json:21]: why do we need these?

File: examples/cloudflare-workers/index.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+console.log("Hello via Bun!");
```
Comment (CahidArda) [inline examples/cloudflare-workers/index.ts:1]: is this needed?

File: examples/cloudflare-workers/package.json (hunk 2 above)
Comment (fahreddinozcan) [reply examples/cloudflare-workers/package.json]: `bun init` command probably adds them automatically. removed

**Comment by fahreddinozcan** [comment]
Final state actually works, see [here](https://github.com/upstash/ratelimit-js/actions/runs/11272101909)

**Review by ytkimirti — APPROVED** [review]

**Pull request merged**